src/charm_state.py

```
@@ -897,3 +929,22 @@ def from_charm(  # noqa: C901
         cls._store_state(state)
 
         return state
+
+
+def _parse_flavor_label_list(flavor_label_config: str) -> list[FlavorLabel]:
+    """Parse flavor-label config option."""
+    combinations = []
+    for flavor_label in flavor_label_config.split(","):
+        flavor_label_stripped = flavor_label.strip()
+        if not flavor_label_stripped:
+            continue
```

Should we raise an error here? Because the user would have made a mistake in the configuration?

---

Not yet at least, a value of "" in `flavor-label-combinations` will get to that condition. That is,  `"".split(",") == [""]`

---

Well, thinking a bit more, I am also allowing trailing commas and things like that... 

I have updated the code to not allow strange combinations of commas.